polygeist/lib/Dialect/Polygeist/Transforms/ArgumentPromotion.cpp

```diff
@@ -609,19 +609,10 @@ bool ArgumentPromotionPass::isCandidateCallable(
     return false;
   }
 
-  // Ensure all the call sites for this function are either in a GPU kernel or
-  // in a function that is called directly by a GPU kernel.
-  // TODO: Could generalize by checking that the call chain from the GPU kernel
-  // are all candidates.
-  Optional<unsigned> maxDepth =
-      funcKernelInfo.getMaxDepthFromAnyGPUKernel(funcOp);
-  assert(maxDepth.has_value() &&
-         "Expecting func to be called from a GPU kernel");
-  assert(maxDepth.value() != 0 && "Expecting func is not itself a GPU kernel");
-  if (maxDepth.value() > 2) {
+  // Check candidate is a SYCL KernelObjFunction.
+  if (!funcKernelInfo.isKernelFuncObjFunction(funcOp)) {
```

Comment:
Why `isKernelFuncObjFunction` cannot be `const`?

Reply:
It's still used in LoopInternalization